common/test/acceptance/tests/lms/test_lms_courseware.py

```diff
@@ -542,3 +544,171 @@ def test_courseware_a11y(self):
         self.courseware_page.a11y_audit.config.set_scope(
             include=['div.sequence-nav'])
         self.courseware_page.a11y_audit.check_for_accessibility_errors()
+
+
+@ddt.ddt
+class ProblemStateOnNavigationTest(UniqueCourseTest):
+    """
+    Test courseware with problems in multiple verticals
+    """
+    USERNAME = "STUDENT_TESTER"
+    EMAIL = "student101@example.com"
+
+    problem1_name = 'MULTIPLE CHOICE TEST PROBLEM 1'
+    problem2_name = 'MULTIPLE CHOICE TEST PROBLEM 2'
+
+    def setUp(self):
+        super(ProblemStateOnNavigationTest, self).setUp()
+
+        self.courseware_page = CoursewarePage(self.browser, self.course_id)
+
+        # Install a course with section, tabs and multiple choice problems.
+        course_fix = CourseFixture(
+            self.course_info['org'], self.course_info['number'],
+            self.course_info['run'], self.course_info['display_name']
+        )
+
+        course_fix.add_children(
+            XBlockFixtureDesc('chapter', 'Test Section 1').add_children(
+                XBlockFixtureDesc('sequential', 'Test Subsection 1,1').add_children(
+                    self.get_multiple_choice_problem(self.problem1_name),
+                    self.get_multiple_choice_problem(self.problem2_name),
+                ),
+            ),
+        ).install()
+
+        # Auto-auth register for the course.
+        AutoAuthPage(
+            self.browser, username=self.USERNAME, email=self.EMAIL,
+            course_id=self.course_id, staff=False
+        ).visit()
+
+        self.courseware_page.visit()
+        self.problem_page = ProblemPage(self.browser)
+
+    def get_multiple_choice_problem(self, problem_name):
```

Comment:
But the problem descriptor does not exist until this function is called, right? In my mind, `get` is used only if you're retrieving a thing that already exists. For example, if you were using the django ORM to make a new object that you later save, you wouldn't call `Model.objects.get()`. You create a new model instance, then save it.